## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 451454 --additional-package pkgsMusl.openssl --additional-package pkgsStatic.openssl`
Commit: `c2e5ce46296c5ffd7d0342d9353f853e337ec7bf`

---
### `x86_64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>lixStatic</li>
    <li>lixStatic.dev</li>
    <li>nixStatic</li>
    <li>nixStatic.dev</li>
    <li>nixStatic.doc</li>
    <li>nixStatic.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>pkgsMusl.openssl</li>
    <li>pkgsMusl.openssl.bin</li>
    <li>pkgsMusl.openssl.debug</li>
    <li>pkgsMusl.openssl.dev</li>
    <li>pkgsMusl.openssl.doc</li>
    <li>pkgsMusl.openssl.man</li>
    <li>pkgsStatic.openssl</li>
    <li>pkgsStatic.openssl.bin</li>
    <li>pkgsStatic.openssl.debug</li>
    <li>pkgsStatic.openssl.dev</li>
    <li>pkgsStatic.openssl.doc</li>
    <li>pkgsStatic.openssl.etc</li>
    <li>pkgsStatic.openssl.man</li>
    <li>tests.cross.mbuffer</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>lixStatic</li>
    <li>lixStatic.dev</li>
    <li>nixStatic</li>
    <li>nixStatic.dev</li>
    <li>nixStatic.doc</li>
    <li>nixStatic.man</li>
    <li>pkgsMusl.openssl</li>
    <li>pkgsMusl.openssl.bin</li>
    <li>pkgsMusl.openssl.debug</li>
    <li>pkgsMusl.openssl.dev</li>
    <li>pkgsMusl.openssl.doc</li>
    <li>pkgsMusl.openssl.man</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>pkgsStatic.openssl</li>
    <li>pkgsStatic.openssl.bin</li>
    <li>pkgsStatic.openssl.debug</li>
    <li>pkgsStatic.openssl.dev</li>
    <li>pkgsStatic.openssl.doc</li>
    <li>pkgsStatic.openssl.etc</li>
    <li>pkgsStatic.openssl.man</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>pkgsMusl.openssl</summary>
<pre>ok
99-test_fuzz_x509.t ...................... 
# The results of this test will end up in test-runs/test_fuzz_x509
1..2
ok 1 - require &#x27;../../test/recipes/fuzz.pl&#x27;;
ok 2 # skip No directory ../../fuzz/corpora/x509
ok

Test Summary Report
-------------------
90-test_threads.t                      (Wstat: 256 (exited 1) Tests: 3 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=350, Tests=4503, 924 wallclock secs (13.06 usr  3.66 sys + 717.14 cusr 247.88 csys = 981.74 CPU)
Result: FAIL
make[2]: *** [Makefile:4105: run_tests] Error 1
make[2]: Leaving directory &#x27;/build/openssl-3.6.0&#x27;
make[1]: *** [Makefile:4102: _tests] Error 2
make[1]: Leaving directory &#x27;/build/openssl-3.6.0&#x27;
make: *** [Makefile:4100: tests] Error 2</pre>
</details>
</details>

--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -87,15 +87,7 @@ let
         substituteInPlace Configurations/unix-Makefile.tmpl \
           --replace 'ENGINESDIR=$(libdir)/engines-{- $sover_dirname -}' \
                     'ENGINESDIR=$(OPENSSLDIR)/engines-{- $sover_dirname -}'
-      ''
-      # This test will fail if the error strings between the build libc and host
-      # libc mismatch, e.g. when cross-compiling from glibc to musl
-      +
-        lib.optionalString
-          (finalAttrs.finalPackage.doCheck && stdenv.hostPlatform.libc != stdenv.buildPlatform.libc)
-          ''
-            rm test/recipes/02-test_errstr.t
-          '';
+      '';
 
       outputs = [
         "bin"

--- a/pkgs/development/libraries/openssl/default.nix
+++ b/pkgs/development/libraries/openssl/default.nix
@@ -191,6 +191,11 @@ let
             "--openssldir=/.$(etc)/etc/ssl"
         )
       ]
+      # Tell build system it's cross environment. This allows to skip tests
+      # that would fail when libc is different. Otherwise, run the tests.
+      ++ lib.optional (
+        !lib.systems.equals stdenv.buildPlatform stdenv.hostPlatform
+      ) "--cross-compile-prefix=${lib.getBin stdenv.cc}/bin/"
       ++ lib.optionals withCryptodev [
         "-DHAVE_CRYPTODEV"
         "-DUSE_CRYPTODEV_DIGESTS"

